app/scripts/modules/oracle/serverGroup/configure/serverGroupConfiguration.service.js

```diff
@@ -0,0 +1,112 @@
+import {SECURITY_GROUP_READER} from 'core/securityGroup/securityGroupReader.service';
+import {OracleBMCSProviderSettings} from '../../oraclebmcs.settings';
+
+let angular = require('angular');
+
+module.exports = angular.module('spinnaker.oraclebmcs.serverGroup.configure.configuration.service', [
+  ACCOUNT_SERVICE,
+  NETWORK_READ_SERVICE,
+  SUBNET_READ_SERVICE,
+  SECURITY_GROUP_READER
+])
+  .factory('oraclebmcsServerGroupConfigurationService', function($q, networkReader, oraclebmcsSubnetReader, accountService, oraclebmcsImageReader, securityGroupReader) {
+
+    let oracle = 'oraclebmcs';
+
+    let getShapes = (image) => {
+      if (!image || !image.compatibleShapes) {
+        return [];
+      }
+      return _.map(image.compatibleShapes, function(shape) {
+        return { name: shape };
+      });
+    };
+
+    function configureCommand(application, command) {
+      let defaults = command || {};
+      let defaultCredentials = defaults.account || application.defaultCredentials.oraclebmcs || OracleBMCSProviderSettings.defaults.account;
+      let defaultRegion = defaults.region || application.defaultRegions.oraclebmcs || OracleBMCSProviderSettings.defaults.region;
+
+        return $q.all({
+            credentialsKeyedByAccount: accountService.getCredentialsKeyedByAccount(oracle),
+            networks: networkReader.listNetworksByProvider(oracle),
+            subnets: oraclebmcsSubnetReader.listSubnetsByProvider(oracle),
+            images: loadImages(),
+            availDomains: accountService.getAvailabilityZonesForAccountAndRegion(oracle, defaultCredentials, defaultRegion)
+        }).then(function(backingData) {
+            backingData.accounts = _.keys(backingData.credentialsKeyedByAccount);
+            backingData.filtered = {};
+            backingData.filtered.regions = [{name: 'us-phoenix-1'}];
+            backingData.filtered.availabilityDomains = _.map(backingData.availDomains, function(zone) {
+              return {name: zone};
+            });
+
+            backingData.filterSubnets = function() {
+              if(command.vpcId && command.availabilityDomain) {
+                return _.filter(backingData.subnets, {
+                  vcnId: command.vpcId,
+                  availabilityDomain: command.availabilityDomain
+                });
+              }
+              return backingData.subnets;
+            };
+
+            backingData.availabilityDomainOnChange = function() {
+              command.subnetId = null;
+              backingData.seclists = null;
+            };
+
+            backingData.vpcOnChange = function() {
+              command.subnetId = null;
+              backingData.seclists = null;
+            };
+
+            backingData.subnetOnChange = function() {
+              let subnet = _.find(backingData.subnets, {id : command.subnetId});
+
+              securityGroupReader.getAllSecurityGroups().then(function(allSecurityGroups) {
```

Comment:
You might want to load all security groups above and add them to the backing data - that way, you won't have to reload them every time the subnet changes.